### PR TITLE
Remove unused dependencies (found with depcheck)

### PR DESCRIPTION
**Pull request by joliss**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -105,21 +100,18 @@
...
     "escape-string-regexp": "^1.0.3",
     "exists-sync": "0.0.3",
-    "exit": "^0.1.2",
     "express": "^4.12.3",
     "findup": "0.1.5",
     "findup-sync": "^0.2.1",
     "fs-extra": "0.22.1",
     "fs-monitor-stack": "^1.0.2",
     "git-repo-info": "^1.0.4",
-    "github": "^0.2.3",
```
Comment (rwjblue) [inline package.json]: This is used here: https://github.com/ember-cli/ember-cli/blob/master/bin/changelog#L17

File: package.json [hunk 2]
```diff
@@ -105,21 +100,18 @@
     "ember-router-generator": "^1.0.0",
     "escape-string-regexp": "^1.0.3",
     "exists-sync": "0.0.3",
-    "exit": "^0.1.2",
```
Comment (rwjblue) [inline package.json]: This is used here: https://github.com/ember-cli/ember-cli/blob/058528977255d0e366b8a1760668875cfadf9067/bin/ember#L8

**Comment by joliss** [comment]
Thanks @rwjblue - I've readded exit, github, and multiline, and I also double-checked the remaining ones individually by grepping for `require.*package-name`. Should be OK now.

**Comment by joliss** [comment]
Hm, and I guess we can move the github dependency to devDependencies, because it's only used by bin/changelog. I've added a second commit.

**Comment by stefanpenner** [comment]
LGTM

**Comment by stefanpenner** [comment]
looks like legit failures:

```js
  Acceptance: preprocessor-smoke-test
  ✓ addons with standard preprocessors compile correctly (7853ms)
  1) addon registry entries are added in the proper order
      Running: node_modules/ember-cli/bin/ember build in: /home/travis/build/ember-cli/ember-cli/tmp/some-cool-app
Cannot find module 'broccoli-filter'
Error: Cannot find module 'broccoli-filter'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/home/travis/build/ember-cli/ember-cli/.node_modules-tmp/second-dummy-preprocessor/index.js:3:18)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)

```

**Comment by stefanpenner** [comment]
looking now

**Comment by stefanpenner** [comment]
found it: https://github.com/ember-cli/ember-cli/blob/master/tests/fixtures/preprocessor-tests/app-registry-ordering/node_modules/second-dummy-preprocessor/index.js#L3

**Comment by stefanpenner** [comment]
going to work on this [here](https://github.com/ember-cli/ember-cli/pull/4852)

**Comment by joliss** [comment]
Thanks Stef for finishing this up.

Hm, this has been the second time [my `.gitignore`-aware grep](https://github.com/ggreer/the_silver_searcher) has missed stuff in those `node_modules` - I've sent a PR at https://github.com/ember-cli/ember-cli/pull/4857 to unignore them.

**Comment by stefanpenner** [comment]
@joliss ya I was totally baffled. I didn't even realize those files existed :p

**Pull request closed without merging**